types/systemjs/systemjs-tests.ts

```diff
@@ -1,68 +1,81 @@
-import SystemJS = require('systemjs');
+System.import('./hi.js').then((hi) => {
+    hi.someProperty();
+});
 
-SystemJS.config({
-    baseURL: '/app'
+System.import<Hi>('./hi.js').then(hi => {
+    hi.someExport();
 });
 
-SystemJS.import('main.js');
+System.import('./hi.js', 'https://example.com/base/');
 
-SystemJS.config({
-    // 'plugin-traceur' or 'plugin-typescript' or 'babel' or 'traceur' or 'typescript' or false.
-    transpiler: 'plugin-babel',
-    // or traceurOptions or typescriptOptions
-    babelOptions: {
-    }
-});
+System.register(['foo', 'bar'], (_export, _context) => {
+    let foo;
+    let bar;
 
-SystemJS.config({
-    map: {
-        traceur: 'path/to/traceur.js'
-    }
-});
+    return {
+        setters: [
+            module => {
+                foo = module;
+            },
+            module => {
+                bar = module;
+            }
+        ],
+        execute() {
+            _export('a', 'thing');
```

Comment:
The tests fail on these `_export()` calls, even though they used to be working and I have no idea why they're failing now.

The error message I'm getting:

```
Error: /Users/joeldennning/code/DefinitelyTyped/types/systemjs/systemjs-tests.ts:25:21
ERROR: 25:21  expect  TypeScript@next compile error: 
Argument of type '"a"' is not assignable to parameter of type 'never'.
ERROR: 26:21  expect  TypeScript@next compile error: 
Argument of type '"b"' is not assignable to parameter of type 'never'.
ERROR: 27:21  expect  TypeScript@next compile error: 
Argument of type '"c"' is not assignable to parameter of type 'never'.
ERROR: 28:13  expect  TypeScript@next compile error: 
Expected 2 arguments, but got 1.
```

I don't know where the `never` is coming from in that error.

Reply:
Figured this one out -- went back to polymorphic function signature and `typeof`, which solved the problem here.

Reply:
Oh, it's because you don't want a union type, you want an overload list. Looks like this:
```ts
type ExportFn = {
    (exportName: string, value: any): void;
    (exports: object): void;
}; 
```